.github/workflows/build-and-test.yml

```diff
@@ -0,0 +1,233 @@
+name: build-and-test
+
+on: [push, pull_request]
+
+concurrency:
+  group: build-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  windows-test:
+    name: windows-test
+    runs-on: windows-latest
+    steps:
+      - name: Check out the codebase.
+        uses: actions/checkout@v2
+
+      - name: Caching dependency
+        uses: actions/cache@v2
+        with:
+          path: |
+            %LocalAppData%\go-build
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+
+      - name: Set up Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.17.2
+
+      - name: Install golang dependency
+        run: |
+          $ErrorActionPreference = 'Stop'
+          $env:GOPATH = "${env:USERPROFILE}\go"
+          go env -w CGO_ENABLED=0
+          go install github.com/ory/go-acc@v0.2.6
+
+      - name: Unit tests with coverage
+        run: |
+          $ErrorActionPreference = 'Stop'
+          $env:GOPATH = "${env:USERPROFILE}\go"
+          $env:PATH = "$env:PATH;${env:GOPATH}\bin"
+          go-acc ./...
+
+  installer-script-test:
+    name: installer-script-test
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        PACKAGE_TYPE: [ "deb", "rpm" ]
+    env:
+      PYTHON_VERSION: 3.8.5
+      PIP_VERSION: 20.2.4
+      REQUIREMENTS_PATH: "internal/buildscripts/packaging/tests/requirements.txt"
+      RESULT_PATH: "~/testresults"
+    steps:
+      - name: Check out the codebase.
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Check for relevant changes
+        run: |
+          # Only run installer tests for main branch, tags, or if the branch has relevant changes
+          gitRef=(${GITHUB_REF//// })
+          if [[ "${gitRef[1]}" = "pull" ]] ;
+          then
+            export BASE_BRANCH_NAME=${{ github.base_ref }}
+          else
+            export BASE_BRANCH_NAME=${gitRef[2]}
+          fi
+          export HEAD_BRANCH_NAME=${{ github.sha }}
+          if [[ "${BASE_BRANCH_NAME}" != "main" ]] && [[ "${gitRef[1]}" != "tags" ]] && ! .github/scripts/changes-include internal/buildscripts/packaging/installer internal/buildscripts/packaging/tests ; then
+            echo "SKIP_TESTS=yes" >> $GITHUB_ENV
+          else
+            echo "SKIP_TESTS=no" >> $GITHUB_ENV
+          fi
+
+      - name: Setup pyenv
```

Comment:
Can we try https://github.com/actions/setup-python instead of pyenv?

Reply:
Yes, the same thing we can do using setup-python. In CircleCi we are using `pyenv` so I have kept it over here.
If you suggest then we can use `setup-python` also.

Reply:
Change workflow to use `setup-python`.